src/post/Write/SideControls.js

```diff
@@ -0,0 +1,170 @@
+
+  if (!currentBlock) {
+    return editorState;
+  }
+
+  if (currentBlock.getLength() === 0) {
+    if (currentBlock.getType() === newType) {
+      return editorState;
+    }
+
+    const newBlock = currentBlock.merge({
+      type: newType,
+      data: initialData,
+    });
+
+    const newContentState = contentState.merge({
+      blockMap: blockMap.set(key, newBlock),
+      selectionAfter: selectionState,
+    });
+    return EditorState.push(editorState, newContentState, 'change-block-type');
+  }
+  return editorState;
+}
+
+export default class SideControls extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showControls: false
+    };
+  }
+  findNode({ editorState }) {
+    if (!process.env.IS_BROWSER) return;
+
+    const node = getSelectedBlockNode(window); // eslint-disable-line no-undef
+    if (!node) {
+      debug('No node');
+      if (!this.state.showControls) { this.hide(); }
+      return;
+    }
+
+    const contentState = editorState.getCurrentContent();
+    const selectionState = editorState.getSelection();
+    if (!selectionState.isCollapsed() ||
+      selectionState.anchorKey !== selectionState.focusKey) {
+      debug(
+        'Selection state changed to be (collapsed, anchorKey)',
+        selectionState.isCollapsed()
+      );
+      this.hide();
+      return;
+    }
+
+    const block = contentState.getBlockForKey(selectionState.anchorKey);
+    if (block.getLength() > 0) {
+      debug('Block has content, hidding');
+      this.hide();
+      return;
+    }
+
+    this.show(node);
+  }
+
+  show(node) {
+    this.setState({
+      style: {
+        top: node.offsetTop - 4,
+        left: -50,
+      },
+    });
+  }
+
+  hide() {
+    this.setState({ style: null, showControls: false });
+  }
+
+  componentWillReceiveProps(newProps) {
+    setTimeout(() => {
+      this.findNode(newProps);
+    }, 100);
+  }
+
+  onClickUpload = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.fileInput.click();
+  };
+
+  onChangeImage = () => {
+    const fileInput = this.fileInput;
+    const username = this.props.user.name;
+    this.props.uploadFile({ username, fileInput })
+      .then(({ value }) => {
+        this.props.onChange(addNewBlock(
+          this.props.editorState,
+          'atomic:image', {
+            src: value.url,
+          }
+        ));
+      });
+  };
+
+  toggleMenu = (e) => {
+    e.preventDefault();
+    this.setState({ showControls: !this.state.showControls });
+  }
+
+  render() {
+    const showControls = this.state.showControls;
+    return (
+      <div
+        className="SideControls"
+        style={this.state && this.state.style ? this.state.style : {
+          opacity: 0,
+          pointerEvents: 'none',
+        }}
+      >
+        <button onClick={this.toggleMenu}><Icon name={showControls ? 'close' : 'add'} /></button>
+        {showControls && <div className="Controls__menu">
```

Comment:
This style of writing jsx scopes is not clear and not easy to read. This is better:
```
{ showControls &&
  <div className="Controls__menu">
    <button><Icon name="close" /></button>
    <button onMouseDown={this.onClickUpload} type="button">
      <Icon name="add_a_photo" />
    </button>
    <input ref={(c) => { this.fileInput = c; }} onChange={this.onChangeImage} name="file" type="file" />
    <button><Icon name="remove" /></button>
  </div>
}
```